### DISCUSSION TITLE
Package subpath './public/extractFiles' is not defined by "exports" in /home/damian/wiki/node_modules/extract-files/package.json

### DISCUSSION THREAD

**Discussion by P1xelC0da** [requarks/wiki#6388] [Error / Bug Report] [answered]
I am running into an issue with wiki on a clean installation. I made it past the system setup, and that's where things took a turn. Below is the error.
```
1|Wiki  | 2023-04-26T22:07:20.004Z [MASTER] error: Package subpath './public/extractFiles' is not defined by "exports" in /home/damian/Wiki/node_modules/extract-files/package.json
PM2     | App [Wiki:1] exited with code [1] via signal [SIGINT]
```

I am running Ubuntu 22.04 LTS, Node.js 18.16.0 and npm 9.5.1.

**Comment by mzch** [comment]
Quick hack:
```diff
--- node_modules/extract-files/package.json.bak 2023-04-30 07:36:31.456340637 +0900
+++ node_modules/extract-files/package.json     2023-04-30 07:41:29.544318756 +0900
@@ -35,6 +35,9 @@
       "require": "./public/index.js"
     },
     "./public/": "./public/",
+    "./public/extractFiles": "./public/extractFiles.js",
+    "./public/isExtractableFile": "./public/isExtractableFile.js",
+    "./public/ReactNativeFile": "./public/ReactNativeFile.js",
     "./package": "./package.json",
     "./package.json": "./package.json"
   },
```

**Comment by P1xelC0da** [comment]
Thank you. I don’t see the point in this ever being used in production 
with the workarounds I have to do just to get it running on a clean 
installation.

SMH.

---
Damian Freeman
Executive Director of Engineering, Chair of the Board of Directors
Devsaurus

**Comment by NGPixel** [comment, marked as answer]
Please read the [requirements](https://docs.requarks.io/install/requirements#nodejs).

You're using an unsupported Node.js version.

**Comment by P1xelC0da** [comment]
I find it weird you support old versions of Node, but cannot support 
LTS. You know how long the LTS has been out for?

---
Damian Freeman
Executive Director of Engineering, Chair of the Board of Directors
Devsaurus

**Reply by NGPixel** [reply]
As stated in the docs, the upcoming v3 will support newer versions.
There's a breaking change in Node 18 that is not trivial to fix without refactoring large sections of code.
